### ISSUE TITLE
Deleting Postgres Flex timeout and successive error

### ISSUE THREAD

**Issue by jebreuer** [stackitcloud/terraform-provider-stackit#290]
When deleting Postgres Flex instances I always run into a timeout:
```
stackit_postgresflex_instance.tim-dev-db: Still destroying... [id=e1c540fe-97de-4fa0-a244-c640c98e6021,effa1a1e-2105-4e38-9ab1-a1e865f5319d, 15m0s elapsed]
╷
│ Error: Error deleting instance
│
│ Instance deletion waiting: WaitWithContext() has timed out
╵
```

A successive `terraform destroy` yields:
```
stackit_postgresflex_instance.tim-dev-db: Destroying... [id=e1c540fe-97de-4fa0-a244-c640c98e6021,effa1a1e-2105-4e38-9ab1-a1e865f5319d]
╷
│ Error: Error deleting instance
│
│ Calling API: 400 Bad Request, status code 400, Body: {"message":"instance already marked for deletion use force to deleted it","code":400,"type":"Delete"}
│
│
╵
```

After that I need to "terraform state rm" the resource to be able to continue working.

**Expected behavior:**
Return in a timely manner with a proper state.

If it's just a "mark for deletion" function then I expect it to yield as soon as the marking is finished (which shouldn't take 15 minutes, IMHO).

**Comment by vicentepinto98** [comment]
Hi @jebreuer, 
In the provider we always wait for the operation to finish successfully, so that in this case we are sure that the instance was deleted successfully. 

It seems the timeout is too short for this particular operation, we will increase it. 

We also plan to add support for setting a custom timeout on the resources in the future.

**Comment by vicentepinto98** [comment]
The fix is available on the [v0.14.2](https://github.com/stackitcloud/terraform-provider-stackit/releases/tag/v0.14.2) release of the provider.
